### PR TITLE
Numbers concept and Assembly Line exercise

### PR DESCRIPTION
**Pull request by glennj**
This reworks an older concept "types" I had written weeks ago and focuses on "numbers".

The exercise was taken from ruby.

### CODE REVIEW

File: concepts/numbers/about.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# About
+
+## Numbers and numeric operators
+
+All numbers, whether integers or otherwise, are represented as C `double` type: IEEE754 double precision floating point numbers.
+This limits us to 53 bits of precision.
+
+The usual operators are available to work with numbers
```
Comment (IsaacG) [inline concepts/numbers/about.md:8]:
```suggestion
The usual operators are available to use with numbers:
```

File: concepts/numbers/about.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  For one-input functions, pipe the value into the function
+
+  ```sh
+  $ jq -n '(1 | atan) * 4'
+  3.141592653589793
+  ```
+
+  For two-input functions, the functions will ignore input and expect the
+  inputs as parameters
```
Comment (IsaacG) [inline concepts/numbers/about.md:22]: Is it work a mention that `;` is the param-separator?

File: concepts/numbers/about.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+```sh
+$ jq -n 'pow(2;53) as $n | [$n, $n+1]'
+[
+  9007199254740992,
+  9007199254740992
+]
+```
+
+Those numbers should not be equal. 
+That's the loss of precision.
```
Comment (IsaacG) [inline concepts/numbers/about.md:54]:
```suggestion
That is the loss of precision.
```

File: concepts/numbers/about.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+~~~~
+
+<!-- prettier-ignore-end -->
+
+## Conditional Expressions
+
+`jq` uses an [`if-then-else` expression][if-then-else] for conditional expressions.
+As an _expression_, it is placed in a pipeline.
+
+Then syntax is: `if A then B else C end`.
```
Comment (IsaacG) [inline concepts/numbers/about.md:65]:
```suggestion
The syntax is: `if CONDITION then TRUE_VALUE else FALSE_VALUE end`.
```

File: exercises/concept/assembly-line/.meta/design.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to teach the student how the concept of numbers is implemented in jq.
```
Comment (IsaacG) [inline exercises/concept/assembly-line/.meta/design.md:5]:
```suggestion
The goal of this exercise is to teach the student how the concept of numbers is implemented in `jq`.
```

**Comment by glennj** [comment]
@IsaacG at this point, the student has not been introduced to functions. Does the stub file look reasonable? Laying it out this way makes the tests easier to write.

**Comment by IsaacG** [comment]
> @IsaacG at this point, the student has not been introduced to functions. Does the stub file look reasonable? Laying it out this way makes the tests easier to write.

Depends on how reliant the student is on Exercism to teach them the language :)

I think a lot of students have some experience with some programming and would be fine with this.

A small number might be relying on Exercism to teach them everything and this might confuse them. But I think it's fine to set it up this way as you provide the definition and use. If you wanted to be "safer", you could add a comment saying, 

"This is how a function is defined. The "def" line should not be changed; the trailing `;` is required."

...

"This line calls the functions defined above; do not change this line."

**Pull request merged**